crates/uv-python/src/discovery.rs

```diff
@@ -1668,9 +1668,19 @@ fn is_windows_store_shim(_path: &Path) -> bool {
 impl PythonVariant {
     fn matches_interpreter(self, interpreter: &Interpreter) -> bool {
         match self {
-            // TODO(zanieb): Right now, we allow debug interpreters to be selected by default for
-            // backwards compatibility, but we may want to change this in the future.
-            Self::Default => !interpreter.gil_disabled(),
+            Self::Default => {
+                // TODO(zanieb): Right now, we allow debug interpreters to be selected by default for
+                // backwards compatibility, but we may want to change this in the future.
+                if (interpreter.python_major(), interpreter.python_minor()) >= (3, 14) {
+                    // For Python 3.14+, the free-threaded build is not considered experimental
+                    // and can satisfy the default variant without opt-in
+                    true
+                } else {
+                    // In Python 3.13 and earlier, the free-threaded build is considered
+                    // experimental and requires explicit opt-in
+                    !interpreter.gil_disabled()
+                }
+            }
             Self::Debug => interpreter.debug_enabled(),
             Self::Freethreaded => interpreter.gil_disabled(),
```

Comment:
Is it intentional that the logic above isn't present on these two lines? If I understand correctly, the implication is `uvx python3.13t` and `uvx python3.13d` will both use a python3.13td if that's the only one you can find?

Reply:
Yeah, that's the same "backwards compatible" idea captured in the TODO that debug builds don't require opt-in.